### PR TITLE
docs: vert.x example

### PR DESCRIPTION
**Pull request by crystall-bitquill**
### Summary

Sample code for using the driver with Vert.x 

### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/VertxExample/README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+1. Go to https://start.vertx.io/.
+2. Select the Vert.x version, Java, and Gradle.
+3. Add the following dependencies: 
+   - JDBC Client
+   - Vert.x Config
+   - Vert.x Web
+4. Generate the project.
+
+## Step 2: Add the required Gradle dependencies
+Your `build.gradle` file should include the following dependencies:
```
Comment (karenc-bq) [inline examples/VertxExample/README.md:45]:
```suggestion
Your `build.gradle.kts` file should include the following dependencies:
```

File: examples/VertxExample/README.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    .put("driverClass", "software.amazon.jdbc.Driver")
+    .put("user", "username")
+    .put("password", "password")
+    .put("minSize", "10")
+    .put("maxSize", "20")
+    .put("testConnectionOnCheckout", "false")
+    .put("testConnectionOnCheckin", "false")
+    .put("idleConnectionTestPeriod", "30")
+    .put("numHelperThreads", "10")
+    .put("max_pool_size", 16);
```
Comment (karenc-bq) [inline examples/VertxExample/README.md:75]:
```suggestion
    .put("max_pool_size", 20);
```

File: examples/VertxExample/README.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
```
Comment (karenc-bq) [inline examples/VertxExample/README.md:88]:
```suggestion
    .put("max_pool_size", 20);
```

File: examples/VertxExample/README.md (hunk 2 above)
Comment (karenc-bq) [reply examples/VertxExample/README.md:75]: What is the difference between maxSize and max_pool_size?

**Review by karenc-bq — APPROVED** [review]

File: examples/VertxExample/README.md (hunk 2 above)
Comment (crystall-bitquill) [reply examples/VertxExample/README.md]: The maxSize and minSize properties were actually from [hibernate specific](https://www.mchange.com/projects/c3p0/#hibernate-specific) properties, they've been removed.

File: examples/VertxExample/README.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    .put("idleConnectionTestPeriod", "30")
+    .put("numHelperThreads", "10")
+    .put("max_pool_size", 20);
+
+JDBCPool write;
+JDBCPool read;
+```
+
+## Step 4: Set up Routing
+To set up the HTTP server, you can first create a router. This tutorial will only walk through creating one route, which will obtain the current instance ID, but the sample code will contain examples of different routes.
```
Comment (aaronchung-bitquill) [inline examples/VertxExample/README.md:91]:
```suggestion
To set up the HTTP server, you can first create a router. This tutorial will only walk through creating one route, which will obtain the instance ID of the current database. However, the sample code will contain examples of different routes.
```

Comment (aaronchung-bitquill) [reply examples/VertxExample/README.md:91]: Nit: Add clarity to what type of "instance" is being referred to.

File: wrapper/src/main/java/software/amazon/jdbc/Driver.java [hunk 5]
```diff
@@ -101,21 +102,22 @@ public Connection connect(final String url, final Properties info) throws SQLExc
...
-      throw new SQLException(Messages.get(
-          "Driver.missingDriver",
-          new Object[] {
-              driverUrl,
-              Collections.list(DriverManager.getDrivers()).stream().map(x -> x.getClass().getName())}), e);
+      final List<String> registeredDrivers = Collections.list(DriverManager.getDrivers())
+          .stream()
+          .map(x -> x.getClass().getName())
+          .collect(Collectors.toList());
+      throw new SQLException(Messages.get("Driver.missingDriver", new Object[] {driverUrl, registeredDrivers}), e);
```
Comment (aaronchung-bitquill) [inline wrapper/src/main/java/software/amazon/jdbc/Driver.java:109]: lol sneaky fix?

Comment (aaronchung-bitquill) [reply wrapper/src/main/java/software/amazon/jdbc/Driver.java:109]: BTW no issue with the changes. Just curious. :)

File: examples/VertxExample/README.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+```gradle
+dependencies {
+    implementation(platform("io.vertx:vertx-stack-depchain:4.4.2"))
+    implementation("io.vertx:vertx-core")
+    implementation("io.vertx:vertx-config")
+    implementation("io.vertx:vertx-jdbc-client")
+    implementation("io.vertx:vertx-web")
+    implementation("com.fasterxml.jackson.core:jackson-databind:2.12.7.1")
+    implementation("org.postgresql:postgresql:42.5.4")
+    implementation(project(":aws-advanced-jdbc-wrapper"))
```
Comment (sergiyvamz) [inline examples/VertxExample/README.md:56]: The following question is a generic one and relates to all code examples. It make sense to refer to `":aws-advanced-jdbc-wrapper"` project when sample code is part of the the driver project. However, if a user wants to grab a sample code and make a separate sample app, such dependency should be replaced with a maven one.

implementation("software.amazon.jdbc:aws-advanced-jdbc-wrapper:latest")

Should we mention that in code examples?

**Review by sergiyvamz — APPROVED** [review]

**Pull request merged**